docs/db2graph/db2graph.rst

```diff
@@ -0,0 +1,253 @@
+
+   | The conversion result will be written to ``edges.txt`` in a newly created directory named ``./output_dir``:
+    
+    .. code-block:: bash
+        
+           $ ls -l .
+           output_dir/
+             edges.txt                       # generated file with sets of triples
+             output.log                          # output log file
+           conf/  
+             config.yaml                         # config file
+             edges_queries.txt             # defines edges_queries    
+          $ cat output_dir/edges.txt
+          column_name_A    relation_name_A_to_B    column_name_B
+          column_name_B    relation_name_B_to_C    column_name_C
+    
+Example Use Case
+""""""""""""""""""""
+
+We use `The Movie Dataset <https://www.kaggle.com/datasets/rounakbanik/the-movies-dataset>`_ from Kaggle as an example to demonstrate a step-by-step walkthrough from loading a dataset into a PostgreSQL database to defining the edge queries and to converting the database into a graph using Db2Graph. Note the following steps assume the database has not been created and Marius has not been installed.
+
+#. | First, create a docker container from the PostgreSQL image. This container will contain all of our work in this example. Note that the password of this container, ``password``, will be the password of the database we create.
+
+    .. code-block:: bash
+    
+       $ docker run --name movies_dataset -e POSTGRES_PASSWORD=password -d postgres:12  
+       $ docker exec -it movies_dataset bash # Attach to the container in interactive mode in bash
+
+   | Create a PostgreSQL database ``test_db`` with the username set to ``postgres`` and the password being ``password``. (Assuming in the root directory)
+    
+       .. code-block:: bash
+    
+        $ psql -U postgres
+        > postgres=# create database test_db; 
+        > postgres=# \q
+
+   | Download `The Movie Dataset <https://www.kaggle.com/datasets/rounakbanik/the-movies-dataset>`_ from Kaggle and load it using `the-movie-database-import <https://github.com/guenthermi/the-movie-database-import.git>`_ script. The script allows for easy import into the PostgreSQL database created in prior steps. Note that we place the downloaded ``archive.zip`` dataset from Kaggle in the ``dataset/`` directory. We skip the downloading step as different ways can be used. 
+    
+       .. code-block:: bash
+    
+        $ git clone https://github.com/guenthermi/the-movie-database-import.git 
+        $ cd the-movie-database-import
+        $ vi db_config.json # update the username, password, host, and db_name fields if applicable. Here, password is changed to 'password' and db_name is 'test_db'
+        $ mkdir dataset/ # create a dataset directory and place the downloaded dataset file in it
+        $ unzip archive.zip # unzip the downloaded dataset file
```

Comment:
Where did this archive.zip come from? The download step is missing.

Reply:
To download this thing I need a Kaggle account? We really should not require users to have a Kaggle account to run an example.

Reply:
I created a Kaggle account and it's not clear how I can download this to a remote machine with something simple like `wget`. I have to download it locally and do a remote copy.

Reply:
archive.zip was the file downloaded from Kaggle, this example assumed the user has downloaded the file locally and did a remote copy like what you did. I agree this is unnecessary but there are a few ways to download from Kaggle, the two main ways are: 1. Log in and download locally or 2. Set up an API token and download through the command line using its API which may further complicate this example. So, we skipped the downloading step and let the user decide (briefly mentioned it in [this step](https://github.com/marius-team/marius/pull/99/files#diff-8c759ca9f66f87dc8a8fa2f6832eb4b2842f59f976c12bcce64df608984da5ceR156) 

I guess due to the difficulty of setup this dataset isn't the best choice. We'll try to find a new one.